apps/sms/js/activity_handler.js

```diff
@@ -229,25 +229,24 @@ var ActivityHandler = {
     //Validate if message still exists before opening message thread
     //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
     if (!message) {
-      return;
+      return Promise.reject(new Error('Message is not defined!'));
     }
 
-    MessageManager.getMessage(message.id).then((message) => {
+    return MessageManager.getMessage(message.id).then((message) => {
```

Comment:
I don't like that I've added too much ```return``` statements here, but I feel that's more correct at the same time